pepper-apis/src/main/java/org/broadinstitute/ddp/filter/RateLimitFilter.java

```diff
@@ -0,0 +1,36 @@
+package org.broadinstitute.ddp.filter;
+
+import java.util.Map;
+import javax.servlet.ServletException;
+
+import org.broadinstitute.ddp.exception.DDPException;
+import org.eclipse.jetty.servlets.DoSFilter;
+import spark.Filter;
+import spark.Request;
+import spark.Response;
+
+public class RateLimitFilter implements Filter {
+
+    private final int maxRequestsPerSecond;
+    private final int burst;
+
+    public RateLimitFilter(int maxRequestsPerSecond, int burst) {
+        this.maxRequestsPerSecond = maxRequestsPerSecond;
+        this.burst = burst;
+    }
+
+    @Override
+    public void handle(Request request, Response response) throws Exception {
+        DoSFilter rateLimitFilter = new DoSFilter();
```

Comment:
`handle()` is called every time a request comes in. Maybe we move the `rateLimitFilter.init()` to the constructor?

Reply:
Are we guaranteed that `handle` is going to be called synchronously in the same thread, or is `DoSFilter` reentrant?

Reply:
Source for `DosFilter` sure seems like it has a lot of stuff around thread safety.  Stateless from what I can tell, with vars striped by the relevant `request` when needed.  Docs don't explicitly claim thread safety, but I'm curious what others think.

Perhaps better to be conservative on this one, init `FilterConfig` in the constructor to save a few microseconds, and then bring `DosFilter` declaration back into `handle()`.  What do others think?

Reply:
When you say "DosFilter declaration:", do you mean the instantiation and the init() call?
I expect that you would want to do this only once at startup. Otherwise context of what requests are being throttled/slowed down etc. would be lost.

Reply:
I'm trying this setup locally and it's not working for me. I think we should move the entire `DoSFilter` and `init()` to the constructor, so it's effectively a "global" object. It looks like the DoSFilter does store state (rate trackers) so if we're `new`-ing up a new DoSFilter on every request, that doesn't seem to make sense (how does it know how many requests we have handled?). After tweaking it locally, it seems to behavior as I will expect (i.e. requests are rejected), which I confirmed with the following test:

<img width="669" alt="Screen Shot 2020-05-08 at 11 05 39 AM" src="https://user-images.githubusercontent.com/5713833/81419390-29864a80-911c-11ea-953e-3388be17579f.png">

Reply:
@yufengwng does it work for you now?